### PR TITLE
fix(codegen): remove version replacement for modules.

### PR DESCRIPTION
**Pull request by zhumin8**
- Remove version replacement for generated modules, these versions should be inferred from parent pom.
- set parent version to project version.
- [x] Also need to adjust relevant placeholder from gapic-generator-java side. https://github.com/googleapis/gapic-generator-java/pull/1099

Using placeholder for parent version for unblocking generation workflow setup. For future improvements:
- Use `revision` to get versions from spring-cloud-gcp parent pom. This will involve changes for all modules with `spring-cloud-gcp` as parent.

### CODE REVIEW

**Review by diegomarquezp — APPROVED** [review]
LGTM

File: generated/pom.xml [hunk 1]
```diff
@@ -12,6 +12,22 @@
   <artifactId>generated-parent</artifactId>
   <packaging>pom</packaging>
 
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.google.cloud</groupId>
```
Comment (emmileaf) [inline generated/pom.xml:18]: What is the context for adding `spring-cloud-gcp-core` as a dependency here?

**Review by emmileaf — APPROVED** [review]

File: generated/pom.xml (hunk 1 above)
Comment (emmileaf) [reply generated/pom.xml:18]: Ah just realized we need it for `Credentials`,`DefaultCredentialsProvider`, etc. - got confused with something else here! Resolving.

Comment (zhumin8) [reply generated/pom.xml:18]: Great catch! I missed the fact that we have `spring-cloud-gcp` as parent pom now. Versions for both these dependencies are already specified by BOMs in [`spring-cloud-gcp/pom`](https://github.com/GoogleCloudPlatform/spring-cloud-gcp/blob/1c5b83dc8d039b8a71c9aa185f4bb1a328d746e9/pom.xml#L82-L96). I'll go ahead and remove the `dependencyManagement` section.

Comment (emmileaf) [reply generated/pom.xml]: Makes sense!

**Review by emmileaf — APPROVED** [review]

**Pull request merged**